ts/Extensions/Drilldown.ts

```diff
@@ -1533,7 +1533,10 @@ ColumnSeries.prototype.animateDrillupTo = function (init?: boolean): void {
                         dataLabel = point.dataLabel;
 
 
-                    if (point.graphic) { // #3407
+                    if (
+                        point.graphic && // #3407
+                        point.visible // Don't show if invisible (#18303)
```

Comment:
I didn't test this, but I think `point.visible` can be `undefined`, in which case the point is actually visible. In other parts of the code base we test for an explicit `false`:

https://github.com/highcharts/highcharts/blob/1f4eea0567fa6ba26fde40694a22441b8d52f822/ts/Core/Series/DataLabel.ts#L234

Please check if this may be the case here too.

Reply:
Thanks! From what I can see, `visible: undefined` for line and scatter chart's points works as you described, but for column and pie, they are still invisible. Moreover, in the case of the pie chart, the legend item is locked then.

Demo: https://jsfiddle.net/BlackLabel/w8zcvgbt/

Reply:
Yes, but there's a difference between the options and the `Point` properties. Anyway, it seems that it is legit to use `point.visible` directly. See https://jsfiddle.net/highcharts/r9m56cgb/